packages/widgetbook_core/lib/src/settings/features/knobs/option_knob.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:flutter/material.dart';
+import 'package:widgetbook_core/src/settings/features/knobs/knob_property.dart';
+import 'package:widgetbook_core/src/settings/widgets/widgets.dart';
+
+String _defaultLabelBuilder<T>(T option) {
+  return option.toString();
+}
+
+class OptionKnob<T> extends StatefulWidget {
+  const OptionKnob({
+    super.key,
+    required this.name,
+    this.description,
+    required this.value,
+    required this.values,
+    String Function(T option)? labelBuilder,
+    this.onChanged,
+  }) : labelBuilder = labelBuilder ?? _defaultLabelBuilder;
+
+  final String name;
+  final String? description;
+  final T value;
+  final List<T> values;
+  // TODO rename
+  // TODO do we need an index parameter here?
+  final String Function(T option) labelBuilder;
+  final void Function(T value)? onChanged;
+
+  @override
+  State<OptionKnob<T>> createState() => _OptionKnobState<T>();
+}
+
+// TODO we can probably make this a Stateless Widget
+class _OptionKnobState<T> extends State<OptionKnob<T>> {
```

Comment:
I'm a bit unhappy with this implementation. For some scenarios it is quite hard to use the new "option" knob, e.g. for color since we cannot give the color a proper display value. Any ideas?

Reply:
@Roaa94